pkg/plugins/plugindef/plugindef.cue

```diff
@@ -414,6 +414,9 @@ schemas: [{
 		externalServiceRegistration: #ExternalServiceRegistration
 
 		#ExternalServiceRegistration: {
+			// AuthProvider defines which authentication provider will be used by the plugin to authenticate against Grafana.
+			// Defaults to "ServiceAccounts"
+			authProvider?: #AuthProvider
```

Comment:
I wonder if `TokenProvider` wouldn't be a better name? :thinking:

Reply:
do we even need this? Can't we infer that if a plugin defines impersonation settings, it will use the OAuth provider and if not it will use service accounts?

It'd be nice for the underlying auth provider to be transparent for plugin devs if possible.